tests/lang_callable/template/twrongmapit.nim

```diff
@@ -30,3 +30,15 @@ import sequtils
 (var i = @[""];i).applyIt(it)
 # now works:
 echo "##", i[0], "##"
+
+block evaluation_order:
```

Comment:
is it worth checking this invariant for the other `xxxit` templates?

Reply:
Yep, it is, but this PR is only meant to fix immediate bug blocking #714. I'd make extending the test case a follow-up PR.